### PR TITLE
fix: eRSD Warnings

### PR DESCRIPTION
**Pull request by angelathe**
# PULL REQUEST

## Summary
- Update FHIR conversion liquid templates to convert the following fields that provide our eRSD warnings info:
    - eICR Processing Status
    - eICR Processing Status Reason
    - eICR Processing Status Reason Details
- Add the (anonymized) 3.1 sample eCR/RR combined XML as a test file for the FHIR converter snapshot test

## Related Issue

Fixes: [#461](https://github.com/CDCgov/dibbs-ecr-viewer/issues/461)

## Acceptance Criteria

The fhir conversion should be updated to match the spec and then the corner case handling in utils/evaluate.ts should be removed.

Note from Angela: Not sure what was being called in `utils/evaluate.ts` because I don't think it exists anymore.

## Additional Information

**Previous FHIR structure**
```
Composition
    --> Extension: 
        --> eRSD Warning entries
```

**Corrected FHIR structure (according to the spec):** 
Note: Please double-check that I did this right!
```
Composition
    --> Extension: 
        --> Reference to Observation: eICR Processing Status
Observation: eICR Processing Status
    --> value: Processing Status
    --> Reference to Observation: eICR Processing Status Reason
Observation: eICR Processing Status Reason
    --> value: Processing Status Reason
    --> component: holds values for eICR Processing Status Reason Details
```

**Reference: RR spec doc:**
- RR Spec: [Link](https://www.hl7.org/implement/standards/product_brief.cfm?product_id=470)
    - Search for: eICR Processing Status, eICR Processing Status Reason, eICR Processing Status Reason Details for more info on how they relate to each other / their value sets

**Reference: FHIR spec docs & example:**
- FHIR Spec: RR eICR Processing Status Extension (Composition) [(Link)](https://hl7.org/fhir/us/ecr/StructureDefinition-rr-eicr-processing-status-extension.html)
- FHIR Spec: RR eICR Processing Status Observation [(Link)](https://hl7.org/fhir/us/ecr/StructureDefinition-rr-eicr-processing-status-observation.html)
- FHIR Spec: RR eICR Processing Status Reason Observation [(Link)](https://hl7.org/fhir/us/ecr/StructureDefinition-rr-eicr-processing-status-reason-observation.html)
- Example bundle (good for seeing FHIR structure): https://hl7.org/fhir/us/ecr/Bundle-bundle-rr-document-one-cond-one-pha.json.html

## Checklist

Run the updated FHIR converter on the following two eCR Examples:

1. Sample eCR 3.1
- Example: eICR was processed **with a warning** (has eRSD warning, reason, details)
- Note: To run this in the FHIR converter locally, you would need the combined eICR/RR XML. This can be grabbed by running `add_rr_data_to_eicr` in `dibbs-ecr-viewer/containers/fhir-converter`, or just message me.

2. `ecr_sample_input_good_with_RR` (eICR & RR already combined)
- Example: eICR was processed (**no eRSD warnings**)
- This can be found in `dibbs-ecr-viewer` in `containers/validation/tests/assets`

### CODE REVIEW

**Comment by angelathe** [comment]
I want to add tests, but we don't have any sample combined eICR/RR data already in `dibbs-fhir-converter`. Wondering if this should just be added later....?

File: data/Templates/eCR/Resource/_ObservationRREICRProcessingStatusReason.liquid [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            {% assign obEntryRelats = ob.entryRelationship | to_array -%}
+            {% for obEntryRelat in obEntryRelats -%}
+                {% if obEntryRelat.observation and obEntryRelat.observation.templateId.root == "2.16.840.1.113883.10.20.15.2.3.32" -%}
+                    {% comment %} eICR Processing Status Reason Details {% endcomment %}
+                    {
+                        "code":
+                        {
+                            {% include "DataType/CodeableConcept" CodeableConcept: obEntryRelat.observation.code -%}
+                        },
+                        "valueString": {{ obEntryRelat.observation.value._ | to_json_string }}
```
Comment (mcmcgrath13) [inline data/Templates/eCR/Resource/_ObservationRREICRProcessingStatusReason.liquid:43]: should/could we use the `ValueHelper` here instead of assuming the value's form? or is this always how it will be?

**Comment by mcmcgrath13** [comment]
> I want to add tests, but we don't have any sample combined eICR/RR data already in `dibbs-fhir-converter` / we need more data generally. Any thoughts on what to do about this / should this just be added later....?

Adding a combined sample for our snapshot testing seems like relatively low effort for high value here. What do you think? We could also add unit tests for some of the observation templates, though there may be less value for effort on that front

File: data/Templates/eCR/Header.liquid [hunk 2]
```diff
@@ -76,9 +76,26 @@
 {% endfor -%}
 
 {% if msg.ClinicalDocument.section.code.code and msg.ClinicalDocument.section.code.code == "88085-6" -%}
+  {% comment %} Reportability Response {% endcomment %}
```
Comment (mcmcgrath13) [inline data/Templates/eCR/Header.liquid:79]: ❤️

File: data/Templates/eCR/Resource/_ObservationRREICRProcessingStatus.liquid [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            {% if obEntryRelat.observation and obEntryRelat.observation.templateId.root == "2.16.840.1.113883.10.20.15.2.3.21" -%}
+                {% comment %} Reference: eICR Processing Status Reason Observation {% endcomment %}
+                {% assign obsId = obEntryRelat.observation | to_json_string | generate_uuid -%}
+                {% assign fullObsId = obsId | prepend: 'Observation/' -%}
+                "hasMember":
+                    [
+                        {
+                            "reference": "{{ fullObsId }}"
+                        },
+                    ],
```
Comment (mcmcgrath13) [inline data/Templates/eCR/Resource/_ObservationRREICRProcessingStatus.liquid:40]: does the fhir converter appropriately combine multiples of these into one `hasMember` with several items? can we make sure we test that?

File: data/Templates/eCR/Resource/_ObservationRREICRProcessingStatusReason.liquid (hunk 1 above)
Comment (angelathe) [reply data/Templates/eCR/Resource/_ObservationRREICRProcessingStatusReason.liquid]: Yes absolutely!

Note though that I changed `"valueString":"{{ value._ }}",` to `"valueString":"{{ value._ | clean_string_from_tabs | escape_special_chars }}",` in `ValueHelper` -- LMK if you don't like this

For the 3.1 sample eCR, the `value._` registers as Invalid JSON -- I needed to either apply `escape_special_chars` or `to_json_string` to remove the errors.

Comment (mcmcgrath13) [reply data/Templates/eCR/Resource/_ObservationRREICRProcessingStatusReason.liquid]: makes sense! we should probably also apply that to the other `valueString` logic branch to be safe

File: data/Templates/eCR/Resource/_ObservationRREICRProcessingStatus.liquid (hunk 3 above)
Comment (angelathe) [reply data/Templates/eCR/Resource/_ObservationRREICRProcessingStatus.liquid:40]: Hm, so I don't think there could be multiple references since according to the spec, 

> "[eICR Processing Status] SHALL contain exactly one [1..1] eICR Processing Status Reason (identifier:urn:hl7ii:2.16.840.1.113883.10.20.15.2.3.21:2017-04-01) (CONF:3315-600)."

What do you think?

Comment (mcmcgrath13) [reply data/Templates/eCR/Resource/_ObservationRREICRProcessingStatus.liquid:40]: I think we should trust the spec then

**Review by mcmcgrath13 — APPROVED** [review]
🚀
